radicle-cli/src/commands/patch/checkout.rs

```diff
@@ -0,0 +1,39 @@
+use crate::terminal as term;
+use anyhow::anyhow;
+use radicle::cob::patch::{self, PatchId};
+use radicle::git::{self, RefStr};
+use radicle::profile::Profile;
+use radicle::storage::git::Repository;
+
+pub fn run(
+    storage: &Repository,
+    profile: &Profile,
+    git_workdir: &git::raw::Repository,
+    patch_id: &PatchId,
+) -> anyhow::Result<()> {
+    let patches = patch::Patches::open(profile.public_key, storage)?;
+    let patch = patches
+        .get(patch_id)?
+        .ok_or_else(|| anyhow!("Patch `{patch_id}` not found"))?;
+
+    let spinner = term::spinner("Performing patch checkout...");
+
+    // Getting the patch obj!
+    let patch_head = patch.head();
+    let obj = git_workdir.revparse_single(&patch_head.to_string())?;
+
+    let def_name = format!("patch/{}", term::format::cob(patch_id));
+    // checkout the patch in a new branch!
+    let commit = git_workdir.find_commit(obj.id())?;
```

Comment:
`patch.head` gives you the `Oid` (the wrapper one that provides serialization, hence the `into` after), so there's no need to `to_string` just parse it again using `revparse_single`. We can simply get the head and the `find_commit`:
```suggestion
    let head = *patch.head();
    let commit = git_workdir.find_commit(head.into());
    let def_name = format!("patch/{}", term::format::cob(patch_id));
    // checkout the patch in a new branch!
```

Reply:
Yes, this is what I meant!